### PR TITLE
changes to dim_cs_access

### PR DESCRIPTION
**Pull request by allison-code-dot-org**
# Description

this PR contains some filtering/column changes to dim_cs_access based on requests from Hannah W. 

Namely, it: 
- does not include columns that are not useful to end users (source and notes columns) 
- excludes territories 
- excludes schools with no grade level data 
- excludes private schools 
- creates a binary column for cs access, grouping HY and Y

### CODE REVIEW

File: dbt/models/marts/access_report/dim_cs_access.sql [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
-    select * 
+    select 
+        access_report_year
+        , nces_school_id
+        , state
+        , school_name
+        , grade_levels
+        , school_type 
+        , case 
+            when teaches_cs_final in ('HY', 'Y') then 1
```
Comment (coryamanda) [inline dbt/models/marts/access_report/dim_cs_access.sql:12]: Just to confirm - Y and HY are yes, HN, E, N, and unknown are all no? I didn't hear that today, so I assume you and she have aligned on that and the logic reflects Hannah's

**Review by coryamanda — COMMENTED** [review]
Two callouts for consideration
- There's a row where the year is listed as 2458. I'm sure just a typo, but wanted to flag.
- Can you please confirm with Hannah the final %s and counts of schools in each bucket of teaches_cs_final for 2023 and 2024? That number we need to match exactly. 

I did a Trevor query to recreate the filters you applied here (on school_type, grade_levels, state) and then looked at HY + Y for 2023 and 2024. I got 60.4% for 2023 and 54.0% for 2024, which I know are not the right numbers. We need to figure out the discrepancy between that and what she has released to the business.

File: dbt/models/marts/access_report/dim_cs_access.sql (hunk 1 above)
Comment (allison-code-dot-org) [reply dbt/models/marts/access_report/dim_cs_access.sql:12]: E is filtered out in the staging layer, and HN and N are all no

File: dbt/models/marts/access_report/dim_cs_access.sql [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+        , state
+        , school_name
+        , grade_levels
+        , school_type 
+        , case 
+            when teaches_cs_final in ('HY', 'Y') then 1 
+            else 0 
+        end                                                     as teaches_cs
     from {{ ref('stg_external_datasets__access_report_review_table') }}
+    where access_report_year != '2024'
```
Comment (coryamanda) [inline dbt/models/marts/access_report/dim_cs_access.sql:16]: do we not want 2024 in here?

**Review by coryamanda — COMMENTED** [review]
Still doesn't match what's in the CS report for 2023: 58.1% vs 57.5% in the report.

**Pull request merged**